File: two_sum/sol.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] nums2=nums.clone();
    	Arrays.sort(nums);
    	int le=0;
    	int ri=nums.length-1;
    	int n1=0,n2=0;
    	while(le<ri){
    		if(nums[le]+nums[ri]==target){
    			n1=nums[le];
    			n2=nums[ri];
    			break;
    		}
    		else if(nums[le]+nums[ri]>target){
    			ri--;
    		}
    		else
    			le++;
    	}
    	int fi1=0;
    	int fi2=0;
    	for(int i=0;i<nums.length;i++){
    		if(nums2[i]==n1){
    			fi1=i;
    			break;
    		}	
    	}
    	for(int j=nums.length-1;j>=0;j--){
    		if(nums2[j]==n2){
    			fi2=j;
    			break;
    		}	
    	}
    	return new int[]{fi1,fi2};
    }
}

File: two_sum/sol.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sorted_nums = sorted(nums)
        low, high = 0, len(nums) - 1
        while low < high:
            if sorted_nums[low] + sorted_nums[high] == target:
                lower, upper = sorted_nums[low], sorted_nums[high]
                break
            elif sorted_nums[low] + sorted_nums[high] > target:
                high -= 1
            else:
                low += 1

#        i, lower_index, upper_index = 0, None, None
#        while (lower_index is None or upper_index is None) and i < len(nums):
#            if lower_index is None and nums[i] == lower:
#                lower_index = i
#            elif upper_index is None and nums[i] == upper:
#                upper_index = i
#            i += 1
        for i in range(len(nums)):
            if nums[i] == lower:
                lower_index = i
                break;

        for j in range(len(nums) - 1, -1, -1):
            if nums[j] == upper:
                upper_index = j
                break;

        return [min(lower_index, upper_index), max(lower_index, upper_index)]

    def twoSumMap(self, nums: List[int], target: int) -> List[int]:
        num_dict = dict()

        for i, num in enumerate(nums):
            if num not in num_dict:
                num_dict[num] = [i]
            else:
                num_dict[num].append(i)

        for i, num in enumerate(nums):
            diff = target - num

            if diff in num_dict:
                if diff == num and len(num_dict[diff]) == 1:
                    continue
                else:
                    return [i, num_dict[diff][0]]
